File: Java/프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        /*
        Map을 사용
        */
        int answer = 0;
        int count = 0;
        
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int num : tangerine){
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        ArrayList<Integer> list = new ArrayList<>(map.values());
        //내림차순으로 정렬 
        Collections.sort(list, Collections.reverseOrder());
        
        int sum = 0;
        for(int a : list){
        //list에 있는 값에서 
            sum = sum + a; 
            count++;
            if(sum >= k){
                break;
                
            }

        }
        
        return count;
    }
}

File: Python/프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    #서로 다른 종류 최솟값
    collect = {}
    for i in tangerine:
        if i not in collect:
            collect[i] = 1
        else:
            collect[i] += 1   
    sort_collect = sorted(collect.values(), reverse=True)
    
    total = 0
    answer = 0
    for i in sort_collect: 
        total += i 
        answer += 1
        if total >= k:
            return answer
    return answer